sysutils/snooze/Portfile

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        leahneukirchen snooze 0.5 v
+github.tarball_from archive
+revision            0
+categories          sysutils
+platforms           darwin
+license             public-domain
+maintainers         nomaintainer
+description         run a command at a particular time
+
+long_description \
+    snooze is a tool for waiting until a particular time and then \
+    running a command. Together with a service supervision system such \
+    as runit, this can be used to replace cron(8).
+
+homepage            https://github.com/leahneukirchen/snooze
+
+checksums           rmd160  e1113ffb0d056c2dc4dcfb100d33ee57fce484f0 \
+                    sha256  d63fde85d9333188bed5996baabd833eaa00842ce117443ffbf8719c094be414 \
+                    size    8008
+
+use_configure       no
```

Comment:
Since the project compiles everything off of purely `make`, without any use of `automake` et al., then one other enhancement would be making use of the `makefile` portgroup:

```
PortSystem          1.0
PortGroup           github   1.0
PortGroup           makefile 1.0
```

The `makefile` portgroup automatically sets the right compiler depending on the macOS version when running `make`, and using it means you can preclude `use_configure no` and `destroot.destdir`.  You would only need to further add:

```
makefile.has_destdir yes
```

Reply:
I think "makefile.has_destdir yes" is already the default set by the make PG - right? So this line is probably not needed either, but you can just check if it works without...

Reply:
Thanks for the review!

The port destroots OK without an explicit `makefile.has_destdir`.

While the [upstream makefile](https://github.com/leahneukirchen/snooze/blob/a7ead9fde1dd81d8fda7c828d2518c4326a8857f/Makefile) does support `DESTROOT`, it defaults to a `PREFIX` of `/usr/local`.  I am unsure how to avoid the ports-filesystems violation without an explicit `destroot.destdir`.  Here is the output I receive without that line:

```
Executing:  cd "/opt/local/var/macports/build/_Users_saj_ports_sysutils_snooze/snooze/work/snooze-0.5" && /usr/bin/make -w install CC="/opt/local/var/macports/build/_Users_saj_ports_sysutils_snooze/snooze/work/compwrap/cc/usr/bin/clang" CXX="/opt/local/var/macports/build/_Users_saj_ports_sysutils_snooze/snooze/work/compwrap/cxx/usr/bin/clang++" OBJC="/opt/local/var/macports/build/_Users_saj_ports_sysutils_snooze/snooze/work/compwrap/objc/usr/bin/clang" OBJCXX="/opt/local/var/macports/build/_Users_saj_ports_sysutils_snooze/snooze/work/compwrap/objcxx/usr/bin/clang++" INSTALL="/usr/bin/install -c" DESTDIR=/opt/local/var/macports/build/_Users_saj_ports_sysutils_snooze/snooze/work/destroot
make: Entering directory `/opt/local/var/macports/build/_Users_saj_ports_sysutils_snooze/snooze/work/snooze-0.5'
mkdir -p /opt/local/var/macports/build/_Users_saj_ports_sysutils_snooze/snooze/work/destroot/usr/local/bin /opt/local/var/macports/build/_Users_saj_ports_sysutils_snooze/snooze/work/destroot/usr/local/share/man/man1
install -m0755 snooze /opt/local/var/macports/build/_Users_saj_ports_sysutils_snooze/snooze/work/destroot/usr/local/bin
install -m0644 snooze.1 /opt/local/var/macports/build/_Users_saj_ports_sysutils_snooze/snooze/work/destroot/usr/local/share/man/man1
make: Leaving directory `/opt/local/var/macports/build/_Users_saj_ports_sysutils_snooze/snooze/work/snooze-0.5'
Warning: The following existing file was hidden from the build system by trace mode:
  /var/root/.CFUserTextEncoding
The following files would have been hidden from the build system by trace mode if they existed:
  /AppleInternal/XBS/.isChrooted
  /BinaryCache
  /BuildSupport/makeProject
  /usr/gnu/include
Warning: violation by /usr
Warning: snooze violates the layout of the ports-filesystems!
Warning: Please fix or indicate this misbehavior (if it is intended), it will be an error in future releases!
```

```
destroot.destdir    DESTDIR=${destroot} PREFIX=${prefix}
```

My intent here was only to provide an explicit override for `PREFIX`.  I was under the impression that it was not possible to _append_ to the default value for the `destroot.destdir` keyword, and so I copied the existing default value -- and set the value as a whole.  Please let me know if there is a way to avoid this redundancy.

Reply:
The `makefile` portgroup should already be doing this, but you can explicitly pass the prefix to `make` with something like:

```
destroot.pre_args-append PREFIX=${prefix}
```

Reply:
Ah, I'm sorry.  I did not read the documentation closely enough.  I see it now:

> Keywords that support `pre_args` and `post_args` are used to assemble command strings together in a row, as described in the reference sections below. But it should be noted that all keyword argument modifiers implicitly support keyword list modifiers.

I have pushed a fixup with `destroot.pre_args-append` as you suggest.  This works.

Perhaps it would be more semantically harmonious to append to `post_args` instead?  The documentation would imply that `pre_args` is intended for make targets, whereas `DESTDIR=...` is supplied using `post_args`.  Extending that latter vector with another make variable might make more sense.  Either approach produces the same end result, of course.

Reply:
I think you should/could patch the Makefile to do `PREFIX?=/usr/local`, while you're making changes I think you should also change it to be: `CFLAGS+=-g -Wall -Wextra -Wwrite-strings` (i.e., append these flags to what MacPorts already sets and remove the optimizations and use MacPorts' default)

Reply:
Latest push includes a patch against the upstream `Makefile`.  Thanks.

Reply:
If you really would like to stick to the convention you'd name the patchfile `patch-Makefile.diff` ;)

Reply:
No worries. :-)  Amended.

Reply:
great! I'll just approved the other CI runs - once this finishes happily I'll merge.